src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectShim.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal interface IProjectShim : IAnalyzerHost
```

Comment:
The type implementing this interface is a very thin layer on top of AbstractProject for forwarding project system events/properties to the workspace host. The subtypes for current C# and VB project systems are called CSharpProjectShim and VisualBasicProject, so I thought calling it a _shim_ would keep it consistent. I personally don't like WorkspaceContext as that doesn't convey that this is the _project system_ representation of the _Project_ type that exists at the Workspaces layer. I thought of "IProjectSystemProject" but wasn't sure about it. Did you have a better name suggestion?